lua/plenary/enum.lua

```diff
@@ -0,0 +1,43 @@
+local Enum = {}
+
+function Enum:__eq(other)
```

Comment:
I don't think that this is very useful because enums will always be different 'types'

Reply:
What do you mean?

Reply:
I am saying it is not really useful to compare the entire enum and more useful to compare each variant. It is better to have a metatable for each variant and then have an `__eq` metamethod for each variant.